### ISSUE TITLE
Make pubsub option emitSelf configurable

### ISSUE THREAD

**Issue by a1300** [libp2p/js-libp2p#401]
First and foremost: Thanks for this great package and the hard work! :+1: 

<br/>

- **Version**: 0.26.0
- **Platform**: Linux Kubuntu-18 4.18.0-25-generic
- **Subsystem**:

#### Description:
Currently is the option **emitSelf** for `pubsub` packages ([libp2p/js-libp2p-floodsub](https://github.com/libp2p/js-libp2p-floodsub) and [ChainSafe/gossipsub-js](https://github.com/ChainSafe/gossipsub-js)) hardcoded in the code.

https://github.com/libp2p/js-libp2p/blob/fd738f9d51e2166861ce8327c8ef2f4eb52a27be/src/pubsub.js#L10

Both `pubsub` packages support the **emitSelf** option. It would be nice to add the option to specify if self-published messages should be received.

Example: 
```diff
export class Bundle extends libp2p {
  constructor(_options) {
    const defaults = {
      modules: {
        transport: [TCP],
        streamMuxer: [Mplex],
        connEncryption: [SECIO],
        peerDiscovery: [Bootstrap],
        dht: DHT,
        pubsub: GossipSub,
      },
      config: {
        pubsub: {
          enabled: true,
+         emitSelf: false,
        },
      },
    };
    super(defaultsDeep(_options, defaults));
  }
```

**Comment by a1300** [comment]
If this feature request is granted I would like to have this issue assigned to me :slightly_smiling_face:

**Comment by jacobheun** [comment]
Yes, we should pass the configuration into the Pubsub initialization. We should account for other Pubsub options as well when doing this, such as the strict signing and verification params https://github.com/libp2p/js-libp2p-pubsub/blob/v0.2.0/src/index.js#L30-L31. I'd expect the whole pubsub config object to get passed.

**Comment by jacobheun** [comment]
Fixed in #404
